x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/controls.tsx

```diff
@@ -0,0 +1,151 @@
+  }, [cy, setSelectedNode]);
+
+  const nodeId = selectedNode?.data('id');
+  const nodeLabel = selectedNode?.data('label');
+  const nodeType = selectedNode?.data('type');
+
+  // Set up Cytoscape event handlers
+  useEffect(() => {
+    const selectHandler: cytoscape.EventHandler = (event) => {
+      setSelectedNode(event.target);
+      setShowFlyout(true);
+    };
+
+    if (cy) {
+      cy.on('select', 'node', selectHandler);
+      cy.on('unselect', 'node', deselect);
+      // cy.on('data viewport', deselect);
+    }
+
+    return () => {
+      if (cy) {
+        cy.removeListener('select', 'node', selectHandler);
+        cy.removeListener('unselect', 'node', deselect);
+        // cy.removeListener('data viewport', undefined, deselect);
+      }
+    };
+  }, [cy, deselect]);
+
+  if (showFlyout === false) {
+    return null;
+  }
+
+  const nodeDataButton =
+    analyticsId !== nodeLabel && nodeType === JOB_MAP_NODE_TYPES.ANALYTICS ? (
+      <EuiButtonEmpty
+        onClick={() => {
+          getNodeData(nodeLabel);
+          setShowFlyout(false);
+        }}
+        iconType="branch"
+      >
+        {i18n.translate('xpack.ml.dataframe.analyticsMap.flyout.fetchRelatedNodesButton', {
+          defaultMessage: 'Fetch related nodes',
+        })}
+      </EuiButtonEmpty>
+    ) : null;
+
+  return (
+    <EuiPortal>
+      <EuiFlyout
+        ownFocus
+        size="m"
+        onClose={() => setShowFlyout(false)}
+        data-test-subj="mlAnalyticsJobMapFlyout"
+      >
+        <EuiFlyoutHeader>
+          <EuiFlexGroup direction="column" gutterSize="xs">
+            <EuiFlexItem grow={false}>
+              <EuiTitle size="s">
+                <h3 data-test-subj="mlDataFrameAnalyticsNodeDetailsTitle">
+                  {i18n.translate('xpack.ml.dataframe.analyticsMap.flyoutHeaderTitle', {
+                    defaultMessage: 'Details for {type} {id}',
+                    values: { id: nodeLabel, type: nodeType },
+                  })}
+                </h3>
+              </EuiTitle>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody>
+          <EuiFlexGroup direction="column" gutterSize="s">
+            <EuiFlexItem grow={false}>
+              <EuiDescriptionList
+                compressed
+                type="column"
+                listItems={
+                  nodeType === 'index-pattern'
+                    ? getListItems(details[nodeId][nodeLabel])
+                    : getListItems(details[nodeId])
+                }
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlyoutBody>
+        <EuiFlyoutFooter>
+          <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+            <EuiFlexItem grow={false}>{nodeDataButton}</EuiFlexItem>
+            {/* <EuiFlexItem grow={false}>
+              <DeleteButton id={nodeLabel} type={nodeType} />
+            </EuiFlexItem> */}
```

Comment:
commented code

Reply:
Implementing deletion functionality in a follow up. Item is in the meta issue 👍